apisix/plugins/openid-connect.lua

```diff
@@ -263,7 +263,22 @@ function _M.rewrite(plugin_conf, ctx)
     end
 
     if not conf.redirect_uri then
-        conf.redirect_uri = ctx.var.request_uri
+        -- NOTE: 'lua-resty-openidc' requires that 'redirect_uri' be
+        --       different from 'uri'.  So default to append the
+        --       '.apisix/redirect' suffix if not configured.
+        local suffix = "/.apisix/redirect"
+        local uri = ctx.var.uri
+        if core.string.has_suffix(uri, suffix) then
+            -- This is the redirection response from the OIDC provider.
+            conf.redirect_uri = uri
+        else
+            if string.sub(uri, -1, -1) == "/" then
+                conf.redirect_uri = string.sub(uri, 1, -2) .. suffix
+            else
+                conf.redirect_uri = uri .. suffix
+            end
+        end
+        core.log.debug("auto set redirect_uri: ", conf.redirect_uri)
```

Comment:
When there is no `conf.redirect_uri`, it looks like we generate `redirect_uri` for each request(except for the real redirect from IdP).

Can we inject a default value for `redirect_uri` when we add the `openid-connect` plugin.

For example:

1. we require that the match condition for a road has to be a prefix match, like `/openid/*`, then redirect_uri defaults to `/openid/.apisix/redirect`.

Listen to what others say.

Reply:
@liweitianux any thoughts about this?

Reply:
I don't think we can set a proper default value for `redirect_uri`, because a plugin can be configured globally, per-route, and per-service.  An exception is that the `openid-connect` plugin is configured for a specific route, so we can determine a `redirect_uri` that's covered by the route.

Please correct me if I'm mistaken.

Reply:
I agree with you.

Reply:
We need a test case to show that not providing `redirect_uri` works fine.

Reply:
I changed the job this year, and I haven't been working with APISIX for more than one year now. I'm unfamiliar with the test framework, and that was the main reason that this PR was lacking a test case.  It would be great if you can rejuvenate this PR.  Thanks.